src/Model.jl

```diff
@@ -1,5 +1,21 @@
 # Contains the physical specification of the model
 
+using Unitful
+import Unitful.uconvert
+using UnitfulAtomic
+
+# Convert a temperature to an energy via the Boltzmann constant.
+function uconvert(a::Unitful.EnergyUnits, x::Unitful.Quantity{T,Unitful.𝚯,U}) where {T <: Real, U <: Unitful.Units}
+    uconvert(a, Unitful.k * x)
+end
+
+# Convert a value to an energy in atomic units.
+# If given a number, assume it is already in atomic units (pass through directly).
+function to_energy(x::Unitful.Quantity)::Real
+    ustrip(uconvert(UnitfulAtomic.Eh_au, x))
+end
+to_energy(x::Real)::Real = x
```

Comment:
Nice idea, but actually no need to be so complicated. Take a look at the `austrip` function from `UnitfulAtomic`. I have never used it, but it seems to do pretty much what we want.